File: ads/2023_1/POO/aula10_26-04-2023/atividade01/CartaoCredito.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Atividade01
{
    public class CartaoCredito : IPagamento
    {
        public void Pagar(double valor)
        {
            Console.WriteLine($"Pagamento realizado com cartão de crédito no valor de R${valor}");
        }
    }
}

File: ads/2023_1/Laboratório de Programação/aula11_12-05-2023/atividade02/src/CartaoCredito.java
class CartaoCredito extends Pagamento {
  private String numeroCartao;
  private String dataValidade;

  public CartaoCredito(String numeroCartao, String dataValidade) {
    this.numeroCartao = numeroCartao;
    this.dataValidade = dataValidade;
  }

  public void realizarPagamento() {
    System.out.println("Pagamento realizado com cartão de crédito " + numeroCartao);
  }

  public void emitirRecibo() {
    System.out.println("Recibo emitido para pagamento com cartão de crédito " + numeroCartao);
  }
}